sensor/common/message/message.go

```diff
@@ -0,0 +1,49 @@
+package message
+
+import (
+	"context"
+
+	"github.com/stackrox/rox/generated/internalapi/central"
+)
+
+// ExpiringMessage is a wrapper on central.MsgFromSensor with the addition of a Context.
+// The context will be cancelled when the message is expired and should no longer be sent
+// to Central.
+type ExpiringMessage struct {
+	*central.MsgFromSensor
+	Context context.Context
+}
```

Comment:
When I started looking at that PR, I immediately thought about using an interface instead of concrete type. Like that:

```go
func (m *networkFlowManager) ResponsesC() <-chan message.ExpiringMessage {
```

and

```suggestion
type ExpiringMessage interface {
	GetMessage() MsgFromSensor
	 IsExpired() bool
	 WaitUntilExpired() <-chan struct{}
}

type ExpiringMessageWithContext struct {
	*central.MsgFromSensor
	Context context.Context
}
```

Have you considered using an interface? Found any downsides of such approach?

An immediate benefit is that we could change the implementation without needing to change all the `ResponsesC()` functions.

Reply:
> Have you considered using an interface? Found any downsides of such approach?

I have considered. And I had an iteration of this code with interface, but after experimenting with both, I have a few reasons why I believe a concrete type makes more sense in this case:

1) **Simplicity of the refactor:** many components are reading from `ResponseC` and then operating on the message. With a composite type, I didn't have to worry about how the message is being used. Since `ExpiringMessage` has the same properties as `MsgFromSensor`, functions like `GetEvent()` still work on `ExpiringMessage`. With an interface, I'd have to change all cases of `msg.GetEvent()` to `msg.GetMessage().GetEvent()`.

2) **Communicate by calling interfaces with concrete structs**: IMO, it makes for a cleaner design if the messages being passed by the components are concrete structs. Interfaces make sense for the components, which can share function signatures and be mocked.

3) **Obvious passing by pointer intent:** as much as this would still be true with an interface, I feel like it's more clear the intent that we are passing a pointer to the message here and being explicit about it.

Reply:
Thanks for providing more insights into that decision 👍